frontend/src/libraries/explorer-wamp/transactions.ts

```diff
@@ -167,7 +172,20 @@ export default class TransactionsApi extends ExplorerApi {
           //
           // Once the above TODO is resolved, we should just move this to TransactionInfo method
           // (i.e. query the information there only for the specific transaction).
-          transaction.actions = transactionExtraInfo.transaction.actions;
+          const _actions = transactionExtraInfo.transaction.actions;
+          let _uniformActions: ActionWrapper[] = [];
+          _actions.map((action: any) => {
+            if (typeof action !== "string") {
+              const kind = Object.keys(action)[0];
+              _uniformActions.push({
+                kind: kind,
+                args: action[kind]
+              });
+            } else {
+              _uniformActions.push({ kind: action, args: {} });
+            }
+          });
+          transaction.actions = _uniformActions;
```

Comment:
Here is a better way to express it:

```ts
          const actions = transactionExtraInfo.transaction.actions;

          transaction.actions = actions.map((action: RpcAction | string) => {
            if (typeof action === "string") {
              return { kind: action, args: {} };
            } else {
              const kind = Object.keys(action)[0];
              return {
                kind,
                args: action[kind]
              };
            }
          });
```

Reply:
tried but has error 
![image](https://user-images.githubusercontent.com/33007476/74977454-945a9c00-53df-11ea-8d90-a6367182824c.png)

Reply:
And what happens if you do `args: action[kind] as RpcAction[keyof RpcAction]`? (is it even valid? Anyway, find the way to force the types working here as the code is much cleaner and easier to understand this way.)

Reply:
changed a little, have a look